hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java

```diff
@@ -856,4 +865,10 @@ public NamedHqlQueryMemento toMemento(String name) {
 				getHints()
 		);
 	}
+
+	@Override
+	public boolean hasQueryExecutionToBeAddedToStatistics() {
+		return addQueryExecutionToStatistics;
```

Comment:
How about this instead?
```suggestion
		return !CRITERIA_HQL_STRING.equals( hqlString );
```

Reply:
done!